external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -115,8 +116,96 @@ def predict(self, X: xr.Dataset) -> xr.Dataset:
                         attrs={"units": X_stacked["specific_humidity"].units + " / s"},
                     ),
                 }
+            ).unstack(SAMPLE_DIM_NAME)
+        return match_prediction_to_input_coords(X, return_ds)
+
+    def dump(self, path: str) -> None:
+        with put_dir(path) as path:
+            if self.model is not None:
+                model_filename = os.path.join(path, self._MODEL_FILENAME)
+                self.model.save(model_filename)
+            with open(os.path.join(path, self._CONFIG_FILENAME), "w") as f:
+                f.write(
+                    yaml.dump(
+                        {
+                            "input_variables": self.input_variables,
+                            "output_variables": self.output_variables,
+                            "output_metadata": self._output_metadata,
+                        }
+                    )
+                )
+
+
+@io.register("all-keras-no-stack")
+class PureKerasNoStackModel(Predictor):
```

Comment:
Is there a reason to have a separate class for the 'no stacking' predictor rather than a single `PureKerasModel` class with an optional boolean arg to not stack and use a different `*_prediction_to_dataset` function depending on the flag? Is it just considered cleaner?

Reply:
I ended up deleting this class since I had to add stacking to this model.